env/plugin-commands-env/src/envUse.ts

```diff
@@ -2,17 +2,39 @@ import { promises as fs } from 'fs'
 import util from 'util'
 import gfs from 'graceful-fs'
 import path from 'path'
+import { findWorkspaceDir } from '@pnpm/find-workspace-dir'
+import { updateWorkspaceManifest } from '@pnpm/workspace.manifest-writer'
+import { readProjectManifest } from '@pnpm/read-project-manifest'
+import { satisfies } from 'semver'
 import { PnpmError } from '@pnpm/error'
+import { logger } from '@pnpm/logger'
 import cmdShim from '@zkochan/cmd-shim'
 import isWindows from 'is-windows'
 import symlinkDir from 'symlink-dir'
 import { type NvmNodeCommandOptions } from './node'
 import { CURRENT_NODE_DIRNAME, getNodeExecPathInBinDir, getNodeExecPathInNodeDir } from './utils'
-import { downloadNodeVersion } from './downloadNodeVersion'
+import { getNodeVersion, downloadNodeVersion } from './downloadNodeVersion'
 
 export async function envUse (opts: NvmNodeCommandOptions, params: string[]): Promise<string> {
   if (!opts.global) {
-    throw new PnpmError('NOT_IMPLEMENTED_YET', '"pnpm env use <version>" can only be used with the "--global" option currently')
+    const workspaceDir = await findWorkspaceDir(process.cwd()) ?? process.cwd()
+    const {manifest} = await readProjectManifest(workspaceDir)
+    const {runtime} = manifest.devEngines ?? {}
+    const runtimeNodeVersion = runtime?.name === 'node' ? runtime?.version : undefined
+
+    const version = params.at(-1) ?? opts.useNodeVersion ?? runtimeNodeVersion ?? 'latest'
+    const {nodeVersion} = manifest.pnpm?.executionEnv ?? await getNodeVersion(opts, version)
+
+    if (!opts.useNodeVersion || params.length) {
+      await updateWorkspaceManifest(workspaceDir, { useNodeVersion: nodeVersion })
```

Comment:
We should probably use executionEnv instead. Or switch to devEngines if that is a standard.